noggin_messages/__init__.py

```diff
@@ -1,9 +1,100 @@
 from fedora_messaging import message
+from fedora_messaging.schema_utils import user_avatar_url
 
 
-class MemberSponsorV1(message.Message):
+class NogginMessage(message.Message):
+    @property
+    def app_name(self):
+        """
+        Return the name of the application that generated the message.
+
+        Returns:
+            the name of the application (fas)
+        """
+        return "fas"
+
+    @property
+    def agent_name(self):
+        """Return the agent's username for this message.
+
+        Returns:
+            The agent's username, or None if the msg has no agent key.
+        """
+        return self.body["msg"]["agent"]
+
+    @property
+    def user_name(self):
+        """Return the username for this message.
+
+        Returns:
+            The username this message is about
+        """
+        return self.body["msg"]["user"]
+
+    @property
+    def agent_avatar(self):
+        """
+        Return a URL to the avatar of the user who caused the action.
+
+        Returns:
+            The URL to the user's avatar, or None if username is None.
+        """
+        return user_avatar_url(self.agent_name)
```

Comment:
That's basically what the default `agent_avatar` property does, so you don't need to subclass it here.